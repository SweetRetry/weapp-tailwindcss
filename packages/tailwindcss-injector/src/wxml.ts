import type { Buffer } from 'node:buffer'
import path from 'node:path'
import fs from 'fs-extra'
import { Parser } from 'htmlparser2'
import MagicString from 'magic-string'

const srcImportTagsMap: Record<string, string[]> = {
  // audio: ['src', 'poster'],
  // video: ['src', 'poster'],
  // image: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxs/01wxs-module.html
  // wxs: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxml/import.html
  import: ['src'],
  include: ['src'],
}

export interface WxmlDep {
  tagName: string
  start: number
  end: number
  quote: string | null | undefined
  name: string
  value: string
  attrs: Record<string, string>
}

// https://github.com/fb55/htmlparser2/issues/1541
// https://developers.weixin.qq.com/miniprogram/dev/framework/view/wxml/event.html
export function processWxml(wxml: string | Buffer) {
  const ms = new MagicString(wxml.toString())
  const deps: WxmlDep[] = []
  let currentTagName = ''
  let importAttrs: undefined | string[]
  let attrs: Record<string, string> = {}

  const parser = new Parser(
    {
      onopentagname(name) {
        currentTagName = name
        importAttrs = srcImportTagsMap[currentTagName]
      },
      onattribute(name, value, quote) {
        attrs[name] = value
        if (importAttrs) {
          for (const attrName of importAttrs) {
            if (attrName === name) {
              deps.push({
                name,
                value,
                quote,
                tagName: currentTagName,
                start: parser.startIndex,
                end: parser.endIndex,
                attrs,
              })
            }
          }
        }
      },
      onclosetag() {
        currentTagName = ''
        attrs = {}
        importAttrs = undefined
      },
    },
    {
      lowerCaseTags: false,
      xmlMode: true,
    },
  )
  parser.write(
    ms.original,
  )
  parser.end()
  return deps
}

export async function getDepFiles(file: string) {
  const set = new Set<string>()

  async function getDep(filepath: string) {
    if (await fs.exists(filepath)) {
      const dirname = path.dirname(filepath)
      set.add(filepath)
      const wxml = await fs.readFile(filepath, 'utf8')
      const deps = await processWxml(wxml)
      for (const { value } of deps) {
        const p = path.resolve(dirname, value)
        if (!set.has(p)) {
          await getDep(p)
        }
      }
    }
  }

  await getDep(file)

  return set
}
