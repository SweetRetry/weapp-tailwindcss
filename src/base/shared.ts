import groupBy from 'lodash.groupby'
import type { UserDefinedOptions } from '@/types'

export function getGroupedEntries<T> (entries: [string, T][], options: Required<UserDefinedOptions>) {
  const { cssMatcher, htmlMatcher, jsMatcher } = options
  const groupedEntries = groupBy(entries, ([file]) => {
    if (cssMatcher(file)) {
      return 'css'
    } else if (htmlMatcher(file)) {
      return 'html'
    } else if (jsMatcher(file)) {
      return 'js'
    } else {
      return 'other'
    }
  })
  return groupedEntries
}
