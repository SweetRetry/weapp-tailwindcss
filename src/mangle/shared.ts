export const mangleClassPrefix = 'MANGLE__'

export const mangleClassSuffix = '__MANGLE'

export const mangleClassRegex = /MANGLE__[a-zA-Z0-9_-]+__MANGLE/g

export function markForMangled (str: string) {
  if (typeof str === 'string' && str) {
    return `${mangleClassPrefix}${str.trim()}${mangleClassSuffix}`
  }
  return str
}
