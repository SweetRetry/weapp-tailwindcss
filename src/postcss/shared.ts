import { escape } from '@/escape'
import { SimpleMappingChars2String } from '@/dic'
// css 中，要多加一个 '\' 来转义
// for raw css selector
// export function cssSelectorReplacer(selector: string, escapeEntries = MappingChars2StringEntries) {
//   return escape(selector, true, escapeEntries).replace(/\\2c /g, dic[','])
// }

export function internalCssSelectorReplacer(selectors: string, map: Record<string, string> = SimpleMappingChars2String) {
  return escape(selectors, {
    map
  })
}
