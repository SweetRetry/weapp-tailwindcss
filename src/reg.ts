import replace from 'regexp-replace'

export const classRegexp = /(?:class|className)=(?:["']\W+\s*(?:\w+)\()?["']([^"]+)['"]/gim

export const tagRegexp = /<[a-z][-a-z]*[a-z]* *([a-z][-a-z]*[a-z]*(?: *= *"(.*?)")?)* *\/? *>/gim

// /[\r\n\s]*<(?:\/)?([^ =>]+)([^>]*?)(?:\/)?>/gim

// export const noClosedTagRegexp = /[\r\n\s]*<([^ =>]+)([^>]*?)(?:\/)?>/gim

export function classStringReplace (str: string, replacement: (string: string, arr?: RegExpExecArray, index?: number, lastIndex?: number) => string) {
  return replace(str, classRegexp, replacement)
}

export function tagStringRegexp (str: string, replacement: (string: string, arr?: RegExpExecArray, index?: number, lastIndex?: number) => string) {
  return replace(str, tagRegexp, replacement)
}

export const variableRegExp = /{{([^{}]*)}}/g

export function variableMatch (original: string) {
  return variableRegExp.exec(original)
}
