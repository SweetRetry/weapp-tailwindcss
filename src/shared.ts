import path from 'path'

export { styleHandler } from './postcss'
export { templeteHandler } from './wxml'
export { jsxHandler } from './jsx'
export const pluginName = 'weapp-tailwindcss-webpack-plugin'

export function getFileName (file: string) {
  return path.basename(file, path.extname(file))
}
