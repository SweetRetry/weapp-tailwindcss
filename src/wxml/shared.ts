import { MappingChars2String as dic } from '@/dic'
import { ICommonReplaceOptions } from '@/types'
import { mangleMark } from '@/mangle/expose'

// export function arbitraryValuesReplacer (str: string) {}

export function replaceWxml (
  original: string,
  options: ICommonReplaceOptions | boolean = {
    keepEOL: false,
    mangle: false
  }
) {
  if (typeof options === 'boolean') {
    options = {
      keepEOL: options,
      mangle: false
    }
  }
  let res = original
    .replace(/\[/g, dic['['])
    .replace(/\]/g, dic[']'])
    .replace(/\(/g, dic['('])
    .replace(/\)/g, dic[')'])
    .replace(/#/g, dic['#']) // hex
    .replace(/!/g, dic['!']) // css !important
    .replace(/\//g, dic['/'])
    .replace(/\./g, dic['.'])
    .replace(/:/g, dic[':'])
    // https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/8
    .replace(/%/g, dic['%'])
    .replace(/,/g, dic[','])
    .replace(/\\/g, dic['\\'])
    .replace(/'/g, dic["'"])
    .replace(/"/g, dic['"'])
    .replace(/\*/g, dic['*'])
    .replace(/&/g, dic['&'])
    .replace(/@/g, dic['@'])
    .replace(/{/g, dic['{'])
    .replace(/}/g, dic['}'])
  if (!options.keepEOL) {
    res = res
      // 去除无用换行符和空格
      .replace(/[\r\n]+/g, '')
  }
  if (options.mangle) {
    res = mangleMark(res, options.classGenerator)
  }
  return res
}
