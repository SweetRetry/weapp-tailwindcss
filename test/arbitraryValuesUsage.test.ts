import { arbitraryValuesMap } from './arbitraryValuesMap'
import { createWxmlAllowClassCharsRegExp } from '@/reg'
import { replaceWxml } from '@/wxml/shared'

describe('all arbitrary values usages', () => {
  describe.each(Object.entries(arbitraryValuesMap))('%s', (title, subTitleObj) => {
    test.each(Object.entries(subTitleObj))('%s', (subTitle, testCases) => {
      testCases.forEach((testCase) => {
        const res = replaceWxml(testCase)
        const valid = createWxmlAllowClassCharsRegExp().test(res)
        // if (valid === false) {
        //   console.log(title + ' ---> ' + subTitle)
        // }
        expect(valid).toBe(true)
        expect(res).toMatchSnapshot()
      })
    })
  })
})
