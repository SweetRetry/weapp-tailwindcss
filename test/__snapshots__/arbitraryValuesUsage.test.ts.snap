// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all arbitrary values usages ArbitraryVariants Selectors 1`] = `"_bl__am__c_nth-child_pl_3_qr__br__c_underline"`;

exports[`all arbitrary values usages ArbitraryVariants Selectors 2`] = `"lg_c__bl__am__c_nth-child_pl_3_qr__br__c_hover_c_underline"`;

exports[`all arbitrary values usages ArbitraryVariants Selectors 3`] = `"_bl__am__p_br__c_mt-4"`;

exports[`all arbitrary values usages ArbitraryVariants Selectors 4`] = `"_bl__at_supports_pl_display_c_grid_qr__br__c_grid"`;

exports[`all arbitrary values usages ArbitraryVariants Selectors 5`] = `"_bl__at_media_pl_any-hover_c_hover_qr__bal__am__c_hover_bar__br__c_opacity-100"`;

exports[`all arbitrary values usages Backgrounds Background Color 1`] = `"bg-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Backgrounds Background Image 1`] = `"bg-_bl_url_pl__q__s_img_s_hero-pattern_d_svg_q__qr__br_"`;

exports[`all arbitrary values usages Backgrounds Background Position 1`] = `"bg-_bl_center_top_1rem_br_"`;

exports[`all arbitrary values usages Backgrounds Background Size 1`] = `"bg-_bl_length_c_200px_100px_br_"`;

exports[`all arbitrary values usages Backgrounds Gradient Color Stops 1`] = `"from-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Border Color 1`] = `"border-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Border Radius 1`] = `"rounded-_bl_12px_br_"`;

exports[`all arbitrary values usages Borders Border Width 1`] = `"border-t-_bl_3px_br_"`;

exports[`all arbitrary values usages Borders Divide Color 1`] = `"divide-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Divide Width 1`] = `"divide-x-_bl_3px_br_"`;

exports[`all arbitrary values usages Borders Outline Color 1`] = `"outline-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Outline Offset 1`] = `"outline-offset-_bl_3px_br_"`;

exports[`all arbitrary values usages Borders Outline Width 1`] = `"outline-_bl_5px_br_"`;

exports[`all arbitrary values usages Borders Ring Color 1`] = `"ring-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Ring Offset Color 1`] = `"ring-offset-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages Borders Ring Offset Width 1`] = `"ring-offset-_bl_10px_br_"`;

exports[`all arbitrary values usages Borders Ring Width 1`] = `"ring-_bl_10px_br_"`;

exports[`all arbitrary values usages Effects Box Shadow 1`] = `"shadow-_bl_inset_0_0_0_0__h_50d71e_br_"`;

exports[`all arbitrary values usages Effects Box Shadow 2`] = `"shadow-_bl_0_35px_60px_-15px_rgba_pl_0_co_0_co_0_co_0_d_3_qr__br_"`;

exports[`all arbitrary values usages Effects Box Shadow Color 1`] = `"shadow-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Effects Opacity 1`] = `"opacity-_bl_0_d_5_br_"`;

exports[`all arbitrary values usages Effects Opacity 2`] = `"opacity-_bl__d_67_br_"`;

exports[`all arbitrary values usages Effects Opacity 3`] = `"opacity-_bl_1_br_"`;

exports[`all arbitrary values usages Filters Backdrop Blur 1`] = `"backdrop-blur-_bl_10px_br_"`;

exports[`all arbitrary values usages Filters Backdrop Brightness 1`] = `"backdrop-brightness-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Brightness 2`] = `"backdrop-brightness-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Contrast 1`] = `"backdrop-contrast-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Contrast 2`] = `"backdrop-contrast-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Contrast 3`] = `"backdrop-contrast-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Backdrop Grayscale 1`] = `"backdrop-grayscale-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Grayscale 2`] = `"backdrop-grayscale-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Grayscale 3`] = `"backdrop-grayscale-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Backdrop Hue Rotate 1`] = `"backdrop-hue-rotate-_bl_90deg_br_"`;

exports[`all arbitrary values usages Filters Backdrop Hue Rotate 2`] = `"backdrop-hue-rotate-_bl_270deg_br_"`;

exports[`all arbitrary values usages Filters Backdrop Invert 1`] = `"backdrop-invert-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Invert 2`] = `"backdrop-invert-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Invert 3`] = `"backdrop-invert-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Backdrop Opacity 1`] = `"backdrop-opacity-_bl_0_d_5_br_"`;

exports[`all arbitrary values usages Filters Backdrop Opacity 2`] = `"backdrop-opacity-_bl__d_67_br_"`;

exports[`all arbitrary values usages Filters Backdrop Opacity 3`] = `"backdrop-opacity-_bl_1_br_"`;

exports[`all arbitrary values usages Filters Backdrop Saturate 1`] = `"backdrop-saturate-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Saturate 2`] = `"backdrop-saturate-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Saturate 3`] = `"backdrop-saturate-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Backdrop Sepia 1`] = `"backdrop-sepia-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Backdrop Sepia 2`] = `"backdrop-sepia-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Backdrop Sepia 3`] = `"backdrop-sepia-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Blur 1`] = `"blur-_bl_10px_br_"`;

exports[`all arbitrary values usages Filters Brightness 1`] = `"brightness-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Brightness 2`] = `"brightness-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Contrast 1`] = `"contrast-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Contrast 2`] = `"contrast-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Contrast 3`] = `"contrast-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Drop Shadow 1`] = `"drop-shadow-_bl_0_0_0_0__h_50d71e_br_"`;

exports[`all arbitrary values usages Filters Drop Shadow 2`] = `"drop-shadow-_bl_0_35px_35px_rgba_pl_0_co_0_co_0_co_0_d_25_qr__br_"`;

exports[`all arbitrary values usages Filters Grayscale 1`] = `"grayscale-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Grayscale 2`] = `"grayscale-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Grayscale 3`] = `"grayscale-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Hue Rotate 1`] = `"hue-rotate-_bl_90deg_br_"`;

exports[`all arbitrary values usages Filters Hue Rotate 2`] = `"hue-rotate-_bl_270deg_br_"`;

exports[`all arbitrary values usages Filters Invert 1`] = `"invert-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Invert 2`] = `"invert-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Invert 3`] = `"invert-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Saturate 1`] = `"saturate-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Saturate 2`] = `"saturate-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Saturate 3`] = `"saturate-_bl__d_25_br_"`;

exports[`all arbitrary values usages Filters Sepia 1`] = `"sepia-_bl_50_p__br_"`;

exports[`all arbitrary values usages Filters Sepia 2`] = `"sepia-_bl_1_d_75_br_"`;

exports[`all arbitrary values usages Filters Sepia 3`] = `"sepia-_bl__d_25_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Flex 1`] = `"flex-_bl_2_2_0_p__br_"`;

exports[`all arbitrary values usages Flexbox & Grid Flex Basis 1`] = `"basis-_bl_14_d_2857143_p__br_"`;

exports[`all arbitrary values usages Flexbox & Grid Flex Grow 1`] = `"grow-_bl_2_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Flex Shrink 1`] = `"shrink-_bl_2_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Gap 1`] = `"gap-_bl_2_d_75rem_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Auto Columns 1`] = `"auto-cols-_bl_minmax_pl_0_co__2fr_qr__br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Auto Rows 1`] = `"auto-rows-_bl_minmax_pl_0_co__2fr_qr__br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Column Start / End 1`] = `"col-_bl_16__s__span_16_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Row Start / End 1`] = `"row-_bl_span_16__s__span_16_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Template Columns 1`] = `"grid-cols-_bl_200px_minmax_pl_900px_co__1fr_qr__100px_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Grid Template Rows 1`] = `"grid-rows-_bl_200px_minmax_pl_900px_co__1fr_qr__100px_br_"`;

exports[`all arbitrary values usages Flexbox & Grid Order 1`] = `"order-_bl_13_br_"`;

exports[`all arbitrary values usages Interactivity Accent Color 1`] = `"accent-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Interactivity Caret Color 1`] = `"caret-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 1`] = `"caret-_bl_10px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 2`] = `"caret-_bl_20px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 3`] = `"caret-_bl_30px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 4`] = `"caret-_bl_40px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 5`] = `"caret-_bl_50px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 6`] = `"caret-_bl_60px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 7`] = `"caret-_bl_70px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 8`] = `"caret-_bl_80px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 9`] = `"caret-_bl_90px_br_"`;

exports[`all arbitrary values usages Interactivity Caret Size 10`] = `"caret-_bl_100px_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 1`] = `"cursor-_bl_url_pl_hand_d_cur_qr__co__pointer_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 2`] = `"cursor-_bl_pointer_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 3`] = `"cursor-_bl_zoom-in_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 4`] = `"cursor-_bl_zoom-out_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 5`] = `"cursor-_bl_grab_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 6`] = `"cursor-_bl_grabbing_br_"`;

exports[`all arbitrary values usages Interactivity Cursor 7`] = `"cursor-_bl_not-allowed_br_"`;

exports[`all arbitrary values usages Interactivity Pointer Events 1`] = `"pointer-events-_bl_auto_br_"`;

exports[`all arbitrary values usages Interactivity Pointer Events 2`] = `"pointer-events-_bl_none_br_"`;

exports[`all arbitrary values usages Interactivity Pointer Events 3`] = `"pointer-events-_bl_box-only_br_"`;

exports[`all arbitrary values usages Interactivity Pointer Events 4`] = `"pointer-events-_bl_box-none_br_"`;

exports[`all arbitrary values usages Interactivity Pointer Events 5`] = `"pointer-events-_bl_all_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Margin 1`] = `"scroll-m-_bl_24rem_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Margin 2`] = `"scroll-margin-_bl_10px_20px_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Margin 3`] = `"scroll-margin-_bl_20px_30px_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Margin 4`] = `"scroll-margin-y-_bl_17rem_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Padding 1`] = `"scroll-p-_bl_24rem_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Padding 2`] = `"scroll-padding-_bl_10px_20px_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Padding 3`] = `"scroll-padding-_bl_20px_30px_br_"`;

exports[`all arbitrary values usages Interactivity Scroll Padding 4`] = `"scroll-padding-y-_bl_17rem_br_"`;

exports[`all arbitrary values usages Interactivity Tap Highlight Color 1`] = `"tap-highlight-color-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Interactivity Tap Highlight Color Opacity 1`] = `"tap-highlight-color-opacity-_bl_0_d_5_br_"`;

exports[`all arbitrary values usages Interactivity Tap Highlight Color Opacity 2`] = `"tap-highlight-color-opacity-_bl__d_67_br_"`;

exports[`all arbitrary values usages Interactivity Tap Highlight Color Opacity 3`] = `"tap-highlight-color-opacity-_bl_1_br_"`;

exports[`all arbitrary values usages Interactivity Touch Action 1`] = `"touch-action-_bl_auto_br_"`;

exports[`all arbitrary values usages Interactivity Touch Action 2`] = `"touch-action-_bl_pan-x_br_"`;

exports[`all arbitrary values usages Interactivity Touch Action 3`] = `"touch-action-_bl_pan-y_br_"`;

exports[`all arbitrary values usages Interactivity Touch Action 4`] = `"touch-action-_bl_pan-x-y_br_"`;

exports[`all arbitrary values usages Interactivity Touch Action 5`] = `"touch-action-_bl_none_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 1`] = `"touch-callout-_bl_none_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 2`] = `"touch-callout-_bl_default_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 3`] = `"touch-callout-_bl_none_default_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 4`] = `"touch-callout-_bl_default_none_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 5`] = `"touch-callout-_bl_none_default_auto_br_"`;

exports[`all arbitrary values usages Interactivity Touch Callout 6`] = `"touch-callout-_bl_auto_none_default_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 1`] = `"user-drag-_bl_auto_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 2`] = `"user-drag-_bl_element_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 3`] = `"user-drag-_bl_element_none_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 4`] = `"user-drag-_bl_none_element_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 5`] = `"user-drag-_bl_element_auto_br_"`;

exports[`all arbitrary values usages Interactivity User Drag 6`] = `"user-drag-_bl_auto_element_br_"`;

exports[`all arbitrary values usages Interactivity User Select 1`] = `"user-select-_bl_none_br_"`;

exports[`all arbitrary values usages Interactivity User Select 2`] = `"user-select-_bl_text_br_"`;

exports[`all arbitrary values usages Interactivity User Select 3`] = `"user-select-_bl_all_br_"`;

exports[`all arbitrary values usages Interactivity Will Change 1`] = `"will-change-_bl_top_co_left_br_"`;

exports[`all arbitrary values usages Layout Aspect Ratio 1`] = `"aspect-_bl_4_s_3_br_"`;

exports[`all arbitrary values usages Layout Columns 1`] = `"columns-_bl_10rem_br_"`;

exports[`all arbitrary values usages Layout Object Position 1`] = `"object-_bl_center_bottom_br_"`;

exports[`all arbitrary values usages Layout Top / Right / Bottom / Left 1`] = `"top-_bl_3px_br_"`;

exports[`all arbitrary values usages Layout Z-Index 1`] = `"z-_bl_100_br_"`;

exports[`all arbitrary values usages SVG Fill 1`] = `"fill-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages SVG Stroke 1`] = `"stroke-_bl__h_243c5a_br_"`;

exports[`all arbitrary values usages SVG Stroke Width 1`] = `"stroke-_bl_2px_br_"`;

exports[`all arbitrary values usages Sizing Height 1`] = `"h-_bl_32rem_br_"`;

exports[`all arbitrary values usages Sizing Max-Height 1`] = `"max-h-_bl_32rem_br_"`;

exports[`all arbitrary values usages Sizing Max-Width 1`] = `"max-w-_bl_50_p__br_"`;

exports[`all arbitrary values usages Sizing Min-Height 1`] = `"min-h-_bl_50_p__br_"`;

exports[`all arbitrary values usages Sizing Min-Width 1`] = `"min-w-_bl_50_p__br_"`;

exports[`all arbitrary values usages Sizing Width 1`] = `"w-_bl_32rem_br_"`;

exports[`all arbitrary values usages Spacing Margin 1`] = `"m-_bl_5px_br_"`;

exports[`all arbitrary values usages Spacing Padding 1`] = `"p-_bl_5px_br_"`;

exports[`all arbitrary values usages Spacing Space Between 1`] = `"space-y-_bl_5px_br_"`;

exports[`all arbitrary values usages Transforms Rotate 1`] = `"rotate-_bl_90deg_br_"`;

exports[`all arbitrary values usages Transforms Rotate 2`] = `"rotate-_bl_180deg_br_"`;

exports[`all arbitrary values usages Transforms Scale 1`] = `"scale-_bl_1_d_5_br_"`;

exports[`all arbitrary values usages Transforms Scale 2`] = `"scale-_bl_2_br_"`;

exports[`all arbitrary values usages Transforms Skew 1`] = `"skew-_bl_10deg_br_"`;

exports[`all arbitrary values usages Transforms Skew 2`] = `"skew-_bl_20deg_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 1`] = `"origin-_bl_center_center_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 2`] = `"origin-_bl_center_top_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 3`] = `"origin-_bl_center_bottom_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 4`] = `"origin-_bl_left_center_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 5`] = `"origin-_bl_left_top_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 6`] = `"origin-_bl_left_bottom_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 7`] = `"origin-_bl_right_center_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 8`] = `"origin-_bl_right_top_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 9`] = `"origin-_bl_right_bottom_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 10`] = `"origin-_bl_top_center_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 11`] = `"origin-_bl_top_left_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 12`] = `"origin-_bl_top_right_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 13`] = `"origin-_bl_bottom_center_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 14`] = `"origin-_bl_bottom_left_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 15`] = `"origin-_bl_bottom_right_br_"`;

exports[`all arbitrary values usages Transforms Transform Origin 16`] = `"origin-_bl_33_p__75_p__br_"`;

exports[`all arbitrary values usages Transforms Translate 1`] = `"translate-_bl_10px_20px_br_"`;

exports[`all arbitrary values usages Transforms Translate 2`] = `"translate-_bl_20px_30px_br_"`;

exports[`all arbitrary values usages Transforms Translate 3`] = `"translate-y-_bl_17rem_br_"`;

exports[`all arbitrary values usages Transitions & Animation Animation 1`] = `"animate-_bl_wiggle_1s_ease-in-out_infinite_br_"`;

exports[`all arbitrary values usages Transitions & Animation Animation 2`] = `"animate-_bl_slide_1s_ease-in-out_infinite_br_"`;

exports[`all arbitrary values usages Transitions & Animation Animation 3`] = `"animate-_bl_slide_1s_ease-in-out_infinite_backwards_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Delay 1`] = `"delay-_bl_2000ms_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Delay 2`] = `"delay-_bl_1s_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Delay 3`] = `"delay-_bl_2s_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Duration 1`] = `"duration-_bl_2000ms_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Duration 2`] = `"duration-_bl_1s_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Duration 3`] = `"duration-_bl_2s_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Property 1`] = `"transition-property-_bl_all_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Property 2`] = `"transition-property-_bl_background-color_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Property 3`] = `"transition-_bl_height_br_"`;

exports[`all arbitrary values usages Transitions & Animation Transition Timing Function 1`] = `"ease-_bl_cubic-bezier_pl_0_d_95_co_0_d_05_co_0_d_795_co_0_d_035_qr__br_"`;

exports[`all arbitrary values usages Typography Content 1`] = `"before_c_content-_bl__q_Hello_World_q__br_"`;

exports[`all arbitrary values usages Typography Font Family 1`] = `"font-_bl__q_Open_Sans_q__br_"`;

exports[`all arbitrary values usages Typography Font Size 1`] = `"text-_bl_14px_br_"`;

exports[`all arbitrary values usages Typography Font Weight 1`] = `"font-_bl_1100_br_"`;

exports[`all arbitrary values usages Typography Letter Spacing 1`] = `"tracking-_bl__d_25em_br_"`;

exports[`all arbitrary values usages Typography Line Height 1`] = `"leading-_bl_3rem_br_"`;

exports[`all arbitrary values usages Typography List Style Type 1`] = `"list-_bl_upper-roman_br_"`;

exports[`all arbitrary values usages Typography Text Color 1`] = `"text-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Typography Text Decoration Color 1`] = `"decoration-_bl__h_50d71e_br_"`;

exports[`all arbitrary values usages Typography Text Decoration Thickness 1`] = `"decoration-_bl_3px_br_"`;

exports[`all arbitrary values usages Typography Text Indent 1`] = `"indent-_bl_50_p__br_"`;

exports[`all arbitrary values usages Typography Text Underline Offset 1`] = `"underline-offset-_bl_3px_br_"`;
